### ISSUE TITLE
[FEATURE] URLs.txt ability to comment out links

### ISSUE THREAD

**Issue by sukeberal** [Jules-WinnfieldX/CyberDropDownloader#253]
As it stands, there seems to be no way to comment out links in the URLs.txt file that I could figure out. This feature would be useful for large galleries that are only occasionally updated that the user may want to only occasionally scrape, but not want to erase totally from URLs.txt. As it is now, I have taken to writing //GALLERYNAME at the end of the url, but this still takes time (and tries to scrape this nonexistent page, so is not ideal.)

Suggestion:
Implement the detection of commented out lines in the part of the script that handles parsing URLs.txt. 

Example URLs.txt:
```
www.normalurl.party/artwebsite/user/44443434                  ￥GongDelphino
￥https://www.normalurl.party/artwebsite/user/3849309      ￥Drawman640
￥ www.normalurl.party/artwebsite/user/554455454             ￥Drawwoman992
￥                                                                                   https://www.normalurl.party/artwebsite/user/69694201337  ￥WideAttack
https://forumsite.city/thread/443445                                    ￥Thread for Simpsons
``` 
What the desired output to the scraper would be:
```
www.normalurl.party/artwebsite/user/44443434 
https://forumsite.city/thread/443445
```
There is probably a better way to do this than  `￥` or any other single character as they might appear in a url and complicate things. (at the very least `#`  is used in mega links so implementing could be more difficult.), so this is only illustrative. (In general there is probably better, more versatile ways to handle URLs in general, but would require a larger reworking of the script so for the time being, supporting commenting out seems the simplest plan of action. I will make a separate issue with some more long-term feature ideas. )

**Comment by Jules-WinnfieldX** [comment]
The issue with this is the current way I fetch links out of the URLs.txt file is such that it can extract links from html code or a clock of text, etc. 

`(?:http.*?)(?=($|\n|\r\n|\r|\s|\"|\[\/URL]|]\[|\[\/img]))` is the current regex expression. I'm not good enough at regex to keep the current requirements and ignore commented out characters.

**Comment by trashkun1978** [comment]
not really sure the best way to add new code and use git, but would this work, @Jules-WinnfieldX ?

you can update your regex_links from base_functions.py to:

`async def regex_links(urls) -> list:
    return [
        URL(x) for x in [
            x if not( x.lstrip().startswith('#') ) and re.match( r"(?:http.*?)(?=($|\n|\r\n|\r|\s|\"|\[/URL]|]\[|\[/img]))", x ) else None for x in urls
        ] if x
    ]`

and you would need to main.py,

from:
`    with open(input_file, "r", encoding="utf8") as f:
        links += await regex_links(f.read())
`

to:
`
with open(input_file, "r", encoding="utf8") as f:
    links = await regex_links( f.read().splitlines() )
`

it works with a mix of URLs, gibberish and spaces in front of URLs. Attached [URLs.txt](https://github.com/Jules-WinnfieldX/CyberDropDownloader/files/9974200/URLs.txt) file.

**Comment by Jules-WinnfieldX** [comment]
This is included in 3.5.28.

**Comment by trashkun1978** [comment]
Cheers @Jules-WinnfieldX!

**Comment by Jules-WinnfieldX** [comment]
Make sure you have updated versions.

**Comment by Jules-WinnfieldX** [comment]
I'd need to see the full urls.txt, it's working on my end.

**Comment by n30liberal** [comment]
Crashes and detects 0 urls if the first link is commented out.

**Comment by Jules-WinnfieldX** [comment]
Fixed in 3.6.34
